background.ts

```diff
@@ -177,8 +177,20 @@ Electron.app.whenReady().then(async() => {
 
     try {
       deploymentProfiles = await readDeploymentProfiles();
-    } catch (err: any) {
-      showErrorDialog('Error in the Deployment Profile', err.toString(), true);
+    } catch (ex: any) {
+      if (ex instanceof DeploymentProfileError) {
+        // Need to do a heuristic to see if we have `--no-modal-errors` on the command-line.
+        // Normally we don't process the command-line arguments before we know what our locked fields are,
+        // but we need to see if we don't want to deal with dialog boxes.
+        if (commandLineArgs.includes('--no-modal-dialogs')) {
+          noModalDialogs = true;
```

Comment:
You can't process the command-line arguments until you've read the deployment profiles in case the user tries to change a locked field. But if there's a problem with the deployment profile loading, you need to check the command-line arguments to see if `--no-modal-dialogs` was specified. The two actions are inter-dependent.

If there's a cleaner way of handling this, it should be done in a separate PR.

Reply:
Ah, I wasn't aware that settings could be passed in via the command line. :+1: